### PR TITLE
Video Indexer and refactoring

### PR DESCRIPTION
**Pull request by uday31in**

### CODE REVIEW

File: .gitignore
Comment (esbran) [inline .gitignore:1]: I dont think you need to check in this file

File: infrastructure/config/dev.tfvars.example
Comment (esbran) [inline infrastructure/config/dev.tfvars.example:1]: Do we need this?

File: infrastructure/terraform/deployment.tf
Comment (esbran) [inline infrastructure/terraform/deployment.tf:1]: We should be consistent on using variables

File: infrastructure/terraform/locals.tf
Comment (esbran) [inline infrastructure/terraform/locals.tf:1]: Do we need this file?

File: infrastructure/terraform/modules/functions/functions.tf
Comment (esbran) [inline infrastructure/terraform/modules/functions/functions.tf:1]: If possible, use we use System Assigned MI

Comment (uday31in) [reply infrastructure/terraform/modules/functions/functions.tf:1]: why do we need System Assigned? IMHO UAI is easier as the RBAC to storage account is handled already. If we got with SI then we need to do the RBAC to ensure SI have access to underlying storage account to download/upload video.

File: infrastructure/terraform/deployment.tf
Comment (uday31in) [reply infrastructure/terraform/deployment.tf:1]: agreed, which ones need to changing?

File: infrastructure/terraform/modules/functions/functions.tf
Comment (uday31in) [reply infrastructure/terraform/modules/functions/functions.tf:1]: changed to SI as UAI is not supported in latest commit

**Review by esbran — DISMISSED** [review]
Please test that this works

**Review by esbran — APPROVED** [review]

**Pull request merged**